application/lib/view/screens/ScreenSelectionScreen.dart

```diff
@@ -99,36 +99,44 @@ class ScreenSelectionScreen extends BaseScreen {
   }
 
   Widget createScreen(Screen screen, int position) {
-    return Container(
-        margin: const EdgeInsets.all(10.0),
-        child: Column(
-          children: <Widget>[
-            Container(
-              margin: const EdgeInsets.only(bottom: 10.0),
-              child: Text('Screen ' + position.toString(),
-                  style:
-                      TextStyle(fontWeight: FontWeight.bold, fontSize: 16.0)),
-            ),
-            Container(
-                decoration: BoxDecoration(
-                    border: Border.all(width: 1),
-                    borderRadius: const BorderRadius.all(Radius.circular(5.0))),
-                child: Column(
-                  children: <Widget>[createScreenInfoToWidgets(screen)],
-                ))
-          ],
-        ));
+    return GestureDetector(
+      onTap: () {
+        Routes.push(contextObject.getOutput(), ScreenEditorScreen(screen));
+      },
+      child: Container(
+          margin: const EdgeInsets.all(10.0),
+          child: Column(
+            children: <Widget>[
+              Container(
+                margin: const EdgeInsets.only(bottom: 10.0),
+                child: Text('Screen ' + position.toString(),
+                    style:
+                        TextStyle(fontWeight: FontWeight.bold, fontSize: 16.0)),
+              ),
+              Container(
+                  decoration: BoxDecoration(
+                      border: Border.all(width: 1),
+                      borderRadius:
+                          const BorderRadius.all(Radius.circular(5.0))),
+                  child: Column(
+                    children: <Widget>[createScreenInfoToWidgets(screen)],
+                  ))
+            ],
+          )),
+    );
   }
 
   // We need to add more widgets in this method.
   Widget createWidgetFromType(Map<String, dynamic> widget) {
     switch (widget['type'].toString().toLowerCase()) {
       case 'text':
-        return Text(widget['value']);
-      case 'flatbutton':
-        return FlatButton(onPressed: null, child: Text(widget['value']));
+        return const Text('Text');
+      case 'label':
+        return const Text('Label');
+      case 'flat button':
+        return const FlatButton(onPressed: null, child: Text('Flat button'));
       default:
-        throw Exception('Widget not found. Either add it or ensure no typos');
+        throw Exception(widget['type'] +' not found. Either add it or ensure no typos');
     }
```

Comment:
Duplicated string (widget types) - I would recommend that you instead of string use a enum - such that misspelling comes a non-issue. And the create a different class or pattern that have all the information about widgets and how to create them - this can then be injected into who ever needs it!